sympy/core/basic.py

```diff
@@ -1525,6 +1525,20 @@ def match(self, pattern, old=False):
         pattern = sympify(pattern)
         return pattern.matches(self, old=old)
 
+
+    def match_between(self, pattern1, pattern2, old=False):
```

Comment:
@asmeurer Yes, I'll shift this into the new matcher which will be added to the rubi module. I'm planning to implement the new matcher soon, now I'm quit busy with my proposals, Once done I'll work on it.